### PR TITLE
Date tabs

### PR DESCRIPTION
**Pull request by Cataran**

### CODE REVIEW

File: public/partials/front-dashboard/delivery-pickup-hours-form.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?php
+$settings = new Pizza_Custom_Features_Delivery_Hours_Settings();
+$options = $settings->get_hours_by_day($day);
```
Comment (Sonyque) [inline public/partials/front-dashboard/delivery-pickup-hours-form.php:3]: How ```@day``` variable appears to be here?

File: public/partials/front-dashboard/week-days-tab.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    <!-- Modal content -->
+    <div class="pizza-custom-features-modal-content">
+        <div>
+            <span id="pizza-custom-features-delivery-hours-modal-close" class="pizza-custom-features-close">&times;</span>
+        </div>
+        <div class="pizza-custom-features-content">
+            <h2>Pick-up and Delivery hours</h2>
+            <div class="tab" id="week-days-tab">
+                <?php foreach ($days as $key => $value) : ?>
+                    <button id="<?= $value ?>" class="tablinks <?= $key == 0 ? "active" : "" ?>"><?= $value ?></button>
```
Comment (Sonyque) [inline public/partials/front-dashboard/week-days-tab.php:13]: It's better to use here a strict condition ```===```

File: public/partials/checkout-delivery-form.php [hunk 3]
```diff
@@ -1,6 +1,9 @@
 <?php
 // Get store pick-up and delivery hours
-$pickup_delivery_hours = get_store_delivery_hours();
+$delivery_hours_settings = new Pizza_Custom_Features_Delivery_Hours_Settings();
+$date_now = date('l');
+$pickup_delivery_hours = $delivery_hours_settings->get_hours_by_day($date_now);
+$display = $delivery_hours_settings->get_display();
```
Comment (Sonyque) [inline public/partials/checkout-delivery-form.php:6]: Just a proposal: might be better to check ```$display``` first and so the rest of actions only if it exist.

File: includes/class-pizza-custom-features-delivery-hours.php [hunk 4]
```diff
@@ -57,24 +61,34 @@ public function enqueue_public_scripts()
...
             check_ajax_referer('my-special-string', 'security');
-            $errors = $this->validate($_POST['data']);
-            if (empty($errors)) {
-                $data = $_POST['data'];
-                if (array_key_exists('display-pickup-delivery', $data)) {
-                    $this->save_options($data);
-                } else {
-                    $data[] = ['display-pickup-delivery', 'off'];
-                    $this->save_options($data);
+            $settings = new Pizza_Custom_Features_Delivery_Hours_Settings();
```
Comment (Sonyque) [inline includes/class-pizza-custom-features-delivery-hours.php:78]: Would be good to create an object only when needed - is POST data exists.

File: includes/functions.php [hunk 5]
```diff
@@ -95,3 +95,26 @@ function remove_delivery_hours_option()
 		delete_option('delivery_hours');
 	}
 }
+if (!function_exists('st_wc_py_get_template')) {
```
Comment (Sonyque) [inline includes/functions.php:76]: I don't quite understand why this line needed. Could you explain?

Comment (Cataran) [reply includes/functions.php]: remove this statement (copy/paste mistake)

File: includes/class-pizza-custom-features-delivery-hours.php (hunk 4 above)
Comment (Cataran) [reply includes/class-pizza-custom-features-delivery-hours.php]: agree

**Pull request merged**